discli

```diff
@@ -121,11 +129,68 @@ class DiscourseHelper:
         os.remove(filename)
 
     def get_post(self, id):
-        post_path = urljoin(self.identity["url"], f"posts/{id}.json")
-        r = requests.get(post_path, params=self.api_auth)
-        if r.status_code == 200:
-            return json.loads(r.text)
-        raise RuntimeError("couldn't get post {}: status code {}", id, r.status_code)
+        response = self.request(f"posts/{id}.json")
+        if response.status_code == 200:
+            return json.loads(response.text)
+        raise RuntimeError("couldn't get post {}: status code {}", id, response.status_code)
+
+    def get_posts_in_category(self):
+        """
+        Get all the posts in a category
+        """
+
+        parser = argparse.ArgumentParser(description='get all posts in category')
+        parser.add_argument("category", help="name of the category to get posts from", type=str)
+        args = parser.parse_args(sys.argv[2:])
+
+        for page in range(1, 20):
+            print(f"Getting page {page} of results...")
+            response = self.request(f"/search.json?q=category%3A{args.category}&page={page}")
+
+            if response.status_code != 200:
+                raise RuntimeError("couldn't get category posts {}: status code {}", id, response.status_code)
+
+            results = response.json()
+
+            if 'topics' not in results:
+                break
+
+            topics = results['topics']
+
+            for topic in topics:
+                if glob.glob(f"{topic['id']}-*.md"):
+                    print(f"File for topic {topic['id']} already exists. Skipping.")
+                    continue
+
+                self.save_topic_as_file(topic["id"])
+
+    def request(self, path):
+        """
+        Use requests to fetch an API URL,
+        padding in the auth token,
+        and retrying if we get a 429 response
+        """
+
+        response = None
+
+        while response is None or response.status_code == 429:
+            if response is not None and response.status_code == 429:
+                seconds = 5
+                error_message = response.json()["errors"][0]
+                seconds_match = re.search("wait (\d+) seconds", error_message)
+
+                if seconds_match:
+                    seconds = int(seconds_match.groups()[0]) + 1
+
+                print(
+                    f"  > 429 from API, waiting {seconds} seconds ... "
+                    f"('{response.json()['errors']}')"
```

Comment:
A 429 status means ["too many requests"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429). Of course, it would be trivial to write "429: Too many requests. Waiting {seconds} seconds". The error message simply says something like "too many requests, please try again in XX seconds".